File: CustomOnboardingIntro/app/src/main/java/nisrulz/github/example/customonboardingintro/IntroAdapter.java
package nisrulz.github.example.customonboardingintro;

import android.graphics.Color;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class IntroAdapter extends FragmentPagerAdapter {

  public IntroAdapter(FragmentManager fm) {
    super(fm);
  }

  @Override public Fragment getItem(int position) {
    switch (position) {
      case 0:
        return IntroFragment.newInstance(Color.parseColor("#03A9F4"), position); // blue
      default:
        return IntroFragment.newInstance(Color.parseColor("#4CAF50"), position); // green
    }
  }

  @Override public int getCount() {
    return 2;
  }
}

File: CustomOnboardingIntro/app/src/main/java/github/nisrulz/example/customonboardingintro/IntroAdapter.kt
package github.nisrulz.example.customonboardingintro

import android.graphics.Color
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class IntroAdapter(fm: FragmentManager) :
    FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> IntroFragment.newInstance(
                Color.parseColor("#03A9F4"),
                position
            ) // blue
            else -> IntroFragment.newInstance(
                Color.parseColor("#4CAF50"),
                position
            ) // green
        }
    }

    override fun getCount(): Int {
        return 2
    }
}